ios/adapters/epson/NetPrinterEpsonAdapter.m

```diff
@@ -1,112 +1,160 @@
+      [printer setReceiveEventDelegate:weakSelf];
+      do {
+        @try {
+          _retryAttempts++;
+          [self connectAndSendAux:host
+                     printRawData:text
+                          success:successCallback
+                             fail:errorCallback
+                       printerObj:printer];
+          _retryAttempts = 3;
+        } @catch (NSException *exception) {
+          if (_retryAttempts >= 3) {
+            @throw exception;
+          }
+          [printer endTransaction];
+          [printer disconnect];
+          [printer clearCommandBuffer];
+          [NSThread sleepForTimeInterval:(3.0f * _retryAttempts)];
         }
+      } while (_retryAttempts < 3);
+
+      long delayInSeconds = 3;
+      long startTime = [[NSDate date] timeIntervalSince1970];
+      long currTime = [[NSDate date] timeIntervalSince1970];
+      while (!_finishedAsyncCall && startTime + delayInSeconds > currTime) {
+        [NSThread sleepForTimeInterval:0.5];
+        currTime = [[NSDate date] timeIntervalSince1970];
+      }
+      if (!_finishedAsyncCall) {
+        [NSException raise:@"Print failed"
+                    format:@"Delegate Function not called"];
+      }
+      [printer endTransaction];
+      [printer disconnect];
+      [printer clearCommandBuffer];
+      [printer setReceiveEventDelegate:nil];
+    } @catch (NSException *exception) {
 
-        _successCallback = successCallback;
-        _errorCallback = errorCallback;
-
-}
-- (void) onPtrReceive:(Epos2Printer *)printerObj code:(int)code status:(Epos2PrinterStatusInfo *)status printJobId:(NSString *)printJobId
-{
-    NSString *errMsg = [EpsonUtils makeErrorMessage:status];
-
-    [printerObj endTransaction];
-    [printerObj clearCommandBuffer];
-    int result = [printerObj disconnect];
-
-    if ([errMsg  isEqual: @""] && result == EPOS2_SUCCESS) {
-        _successCallback != nil ? _successCallback(@[[NSString stringWithFormat:@"Successfuly printed"]]) : nil;
-    } else {
-        _errorCallback != nil ? _errorCallback(@[[NSString stringWithFormat:@"Delegate with ERROR %i and message %@",result,errMsg]]) : nil;
+      [printer endTransaction];
+      int failResult = [printer disconnect];
+      [printer clearCommandBuffer];
+      [printer setReceiveEventDelegate:nil];
+      errorCallback(
+          @[ [NSString stringWithFormat:@"%@ and disconnect code %i",
+                                        exception.reason, failResult] ]);
+    } @finally {
+      _successCallback = nil;
+      _errorCallback = nil;
     }
+  }
+}
 
-    _successCallback = nil;
-    _errorCallback = nil;
-
-    [printerObj setReceiveEventDelegate:nil];
-    return;
+- (void)connectAndSendAux:(NSString *)host
+             printRawData:(NSString *)text
+                  success:(RCTResponseSenderBlock)successCallback
+                     fail:(RCTResponseSenderBlock)errorCallback
+               printerObj:(Epos2Printer *)printer {
```

Comment:
Nothing changed here, only adding printerObj. The formatting has messed up the diff